src/cpu_features/x86.rs

```diff
@@ -35,7 +37,29 @@ impl CpuFeatureLevel {
 
 impl Default for CpuFeatureLevel {
   fn default() -> CpuFeatureLevel {
-    let detected: CpuFeatureLevel = if is_x86_feature_detected!("avx2") {
+    fn avx512_detected() -> bool {
+      is_x86_feature_detected!("avx512bw")
+        && is_x86_feature_detected!("avx512cd")
+        && is_x86_feature_detected!("avx512dq")
+        && is_x86_feature_detected!("avx512f")
+        && is_x86_feature_detected!("avx512vl")
+    }
+    fn avx512icl_detected() -> bool {
+      avx512_detected()
+        // && is_x86_feature_detected!("avx512bitalg")
+        // && is_x86_feature_detected!("avx512clmulqdq")
```

Comment:
Are these commented lines important?

Reply:
Not until Intel releases a cpu that has a set of avx512 extensions that do not match the pre-Icelake and Icelake cpu.

We just need to know that we are on an Icelake (so probe for at least some of the Icelake-introduced extensions) or not.